src/wallet/wallet.cpp

```diff
@@ -2546,6 +2548,9 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
     // all.
     txNew.nLockTime = std::max(0, chainActive.Height() - 10);
 
+    // Set nBlockExpiry to 20 blocks (default) past current blockheight
+    txNew.nBlockExpiry = chainActive.Height() + blockExpiry;
```

Comment:
We should set it to the minimum of this and the next activation height, so that transactions drop out of the mempool automatically on the pre-upgrade branch if they aren't mined before activation. I'll add an API to #2898 for fetching the latter.